### PR TITLE
Tow 39 update opportunity

### PR DESCRIPTION
**Pull request by RYeeAnn**
Updating opportunity by ID.

Run the server and in Postman, enter this in Body -> raw

- PUT: http://127.0.0.1:8000/opportunity/?id=6
{
    "name": "Updated Opportunity Name",
    "time": "2024-08-06T12:00:00Z",
    "description": "Updated Description",
    "location": "Updated Location"
}

- GET: http://127.0.0.1:8000/opportunity/?id=6
To see the updated opportunity information.

### CODE REVIEW

**Review by GagenvirGill — APPROVED** [review]

File: townhall/myapi/views.py [hunk 1]
```diff
@@ -47,9 +50,28 @@ def handle_opportunity_request(self, request):
...
+    
+    # UPDATE Opportunity
+    @action(detail=False, methods=['put'], url_path='opportunity')
+    def handle_opportunity_update(self, request):
+        opportunity_id = self.request.query_params.get('id')
+        if not opportunity_id:
+            return Response({"error": "ID is required"}, status=status.HTTP_400_BAD_REQUEST)
+
+        opportunity_obj = opportunity_services.get_opportunity(id=int(opportunity_id))
+        if not opportunity_obj:
```
Comment (ansel-rangers11) [inline townhall/myapi/views.py:69]: Good error handling

File: townhall/tests/test_opportunity.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def test_update_opportunity(self):
             description = "Planting some flowers in the community center",
             location = "Richmond"
         )
-        townhall_services.OpportunityServices.update_opportunity(update_opportunity_data)
+        townhall_services.OpportunityServices.update_opportunity(id=3, update_opportunity_data=update_opportunity_data)
```
Comment (ansel-rangers11) [inline townhall/tests/test_opportunity.py:71]: I challenge in this PR as learning point to incorporate endpoint unit tests, for example `client.get` or `client.post` or `client.patch` . You can use this article as reference how to do this https://djangostars.com/blog/django-pytest-testing/

Note how they assert status code and the response. Please also make a new file called `test_endpoint.py` for these new set of tests (which inside will have 3 classes corresponding to each model that we have) . No rush on this changes I know its a lot, good luck and ping me if theres any qs

**Review by ansel-rangers11 — CHANGES_REQUESTED** [review]

File: townhall/tests/test_opportunity.py (hunk 2 above)
Comment (RYeeAnn) [reply townhall/tests/test_opportunity.py:71]: @ansel-rangers11 Just committed an update if you could double check if it was done correctly or not. I used pytest to test the endpoints

Comment (ansel-rangers11) [reply townhall/tests/test_opportunity.py:71]: @RYeeAnn yes looks like youve got it! Ill approve once the merge conflicts have been resolved

Comment (RYeeAnn) [reply townhall/tests/test_opportunity.py:71]: @ansel-rangers11 awesome thanks, just resolved the merge conflict

**Review by ansel-rangers11 — APPROVED** [review]

**Pull request merged**